### PR TITLE
pkp/pkp-lib#946 Archive and record counts should exclude disabled arc…

### PR DESCRIPTION
**Pull request by mfelczak**
…hives

### CODE REVIEW

File: classes/record/RecordDAO.inc.php [hunk 1]
```diff
@@ -264,13 +264,27 @@ function getInsertRecordId() {
...
-	function getRecordCount($archiveId = null) {
-		$result =& $this->retrieve(
-			'SELECT COUNT(*) AS count FROM records' . (isset($archiveId)?' WHERE archive_id = ?':''),
-			isset($archiveId)?$archiveId:false
-		);
+	function getRecordCount($archiveId = null, $onlyEnabled = false) {
+		$sql = 'SELECT COUNT(*) AS count
+				FROM records r';
+
+		if ($onlyEnabled) $sql .= ' LEFT JOIN archives a ON r.archive_id = a.archive_id WHERE a.enabled = 1';
```
Comment (asmecher) [inline classes/record/RecordDAO.inc.php]: This is a little convoluted -- can you rewrite this so we don't need the `$archiveId` to be implemented differently for each `$onlyEnabled` case?

**Comment by mfelczak** [comment]
Cleaned-up SQL logic

File: classes/record/RecordDAO.inc.php (hunk 1 above)
Comment (mfelczak) [inline classes/record/RecordDAO.inc.php]: Cleaned-up SQL logic

File: classes/record/RecordDAO.inc.php [hunk 2]
```diff
@@ -264,13 +264,28 @@ function getInsertRecordId() {
...
-	function getRecordCount($archiveId = null) {
-		$result =& $this->retrieve(
-			'SELECT COUNT(*) AS count FROM records' . (isset($archiveId)?' WHERE archive_id = ?':''),
-			isset($archiveId)?$archiveId:false
-		);
+	function getRecordCount($archiveId = null, $onlyEnabled = false) {
+		$sql = 'SELECT COUNT(*) AS count
+				FROM records r';
+
+		if ($onlyEnabled) {
```
Comment (asmecher) [inline classes/record/RecordDAO.inc.php]: I'm thinking something like...
````
$result =& $this->retrieve(
    SELECT COUNT(*) AS count
    FROM records r 
    ' . ($onlyEnabled?', archives a ':'') . '
    WHERE 1=1
    ' . ($archiveId?' AND r.archive_id = ? ':'') . '
    ' . ($onlyEnabled?' AND r.archive_id = a.archive_id ':'') . '
    $archiveId? (int) $archiveId:false
);
````
(Untested -- just for the gist. But this kind of thing helps keep DAOs short and maintainable.)

Comment (mfelczak) [inline classes/record/RecordDAO.inc.php]: tight!

**Comment by mfelczak** [comment]
Further SQL clean-up

**Comment by asmecher** [comment]
Looks good, thanks!

**Pull request merged**